homeassistant/components/simplisafe/__init__.py

```diff
@@ -167,20 +158,24 @@ async def async_setup_entry(hass, config_entry):  # noqa: C901
     client_id = await async_get_client_id(hass)
     websession = aiohttp_client.async_get_clientsession(hass)
 
-    try:
-        api = await API.login_via_token(
-            config_entry.data[CONF_TOKEN], client_id=client_id, session=websession
+    async def async_get_api():
+        """Define a helper to get an authenticated SimpliSafe API object."""
+        return await API.login_via_credentials(
+            config_entry.data[CONF_USERNAME],
+            config_entry.data[CONF_PASSWORD],
+            client_id=client_id,
+            session=websession,
         )
+
+    try:
+        api = await async_get_api()
     except InvalidCredentialsError:
-        LOGGER.error("Invalid credentials provided")
-        return False
+        raise ConfigEntryAuthFailed  # pylint: disable=raise-missing-from
     except SimplipyError as err:
         LOGGER.error("Config entry failed: %s", err)
         raise ConfigEntryNotReady from err
 
-    _async_save_refresh_token(hass, config_entry, api.refresh_token)
```

Comment:
We should still use the refresh token as much as possible. Let's not remove this logic.

Reply:
`simplisafe-python` does this internally (i.e., refreshes the token whenever it can); if that fails, it raises an `InvalidCredentialsError` up here.